research/job_referencing/index.md

```diff
@@ -0,0 +1,217 @@
+
+### Monorepo: Package dependencies
+
+The following is the [RFE](https://github.com/packit/packit/issues/1903) by LecrisUT
+
+```
+packages:
+A:
+depends:
+- B
+- name: C
+rebuild: true
+B: ...
+C: ...
+```
+
+at first I would simplify it like:
+
+```
+packages:
+A:
+depends:
+- B
+- C
+B: ...
+C: ...
+```
+
+I would always rebuild everything for every package but we can enhance the code and if a package has no changes in the given `commit_sha` and it does not depend on changed packages, we can simply copy lines from the previous database pipeline, filling the database as if the jobs were run.
+
+#### The easiest solution I see following what we are already doing for copr builds and tests
+
+We should [create tasks](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/worker/jobs.py#L464-L497) only for those packages that do not depend on other packages.
+When running, as an example, the COPR build for the latest dependent package in the list (C in our example), at the end of the handler we can create the COPR build tasks for package A (if any).
+This can be done for all the kinds of job we directly start at [this lines](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/worker/jobs.py#L464-L497).
+
+We can change `bodhi` jobs in a way that they will do nothing, apart from starting new `bodhi` tasks for other packages, unless they belong to a package not required by another package. Ideally only package A, in this example, should do a `bodhi` update if I am not wrong. Or at least we should be able to skip doing something in this job during the chain of dependencies.
+
+I think we don't need changes in the database schema to be able to implement this solution.
+We need to add the `depends` or `depends_on` key to the schema. I slightly prefer the `depends` key, but not strong opinions on that.
+
+### Request new feature: support side tag for multi package update
+
+This is the [RFE](https://github.com/packit/packit/issues/1870).
+
+I am not really sure I am getting the point here. But they are talking about a multi-package feature and for this reason I assume they need to use the monorepo syntax (otherwise we have no way to reference the packages).
+If they can and want to use the monorepo syntax then I think that the package dependencies solution for the monorepos would solve their problems too. Or am I wrong?
+
+## Drawbacks
+
+I see just a drawback in the previous solutions, the code is getting more and more complex to read (I mean to grasp the execution order) and test.
+I think `Celery` was well suited for tasks which didn't depend on each other, now that dependencies are growing quickly I feel it is like a limit. I would like to be able to have communication between tasks and suspend the execution without exiting.
+I think that a code using `asyncio`, as an example, could be more readable and testable.
```

Comment:
Do you mean to add some asynchronous waiting to have a single task (~code perspective) for a single job (~user perspective)?

Reply:
At least for the more complex jobs like the copr build that depends on a srpm or the test job that depends on a copr build.

Reply:
Thanks, I should have looked at the text and example bellow before commenting..;)